src/execution_plan/optimizations/reduce_scans.c

```diff
@@ -20,6 +31,12 @@ static void _reduceScans(ExecutionPlan *plan, OpBase *scan) {
 	// Check if that alias is already bound by any of the scan's children.
 	for(int i = 0; i < scan->childCount; i ++) {
 		if(ExecutionPlan_LocateOpResolvingAlias(scan->children[i], scanned_alias)) {
+			if(scan->type == OPType_NODE_BY_LABEL_SCAN) {
```

Comment:
We could have also collected index scans and ID seeks, which I expect also contain data that would need to be converted into filters, no?